From 9f0e0b985eeeabd60aab236f99eb104927578253 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Mon, 3 Jun 2019 08:18:17 +0200
Subject: [PATCH] gioio: implement wpath() to fix build on Windows

On Windows the BasicIo class has an additional pure virtual function
called wpath() guarded by EXV_UNICODE_PATH.

GioIo was missing an implementation and that made the build fail on Windows.

Fixes #41
---
 gexiv2/gexiv2-metadata.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/gexiv2/gexiv2-metadata.cpp b/gexiv2/gexiv2-metadata.cpp
index ab8e091..fb32a6b 100644
--- a/gexiv2/gexiv2-metadata.cpp
+++ b/gexiv2/gexiv2-metadata.cpp
@@ -206,6 +206,15 @@ public:
         return "GIO Wrapper";
     }
 
+#ifdef EXV_UNICODE_PATH
+    std::wstring wpath() const {
+        std::string p = path();
+        std::wstring w(p.length(), L' ');
+        std::copy(p.begin(), p.end(), w.begin());
+        return w;
+    }
+#endif
+
 #if EXIV2_TEST_VERSION(0,27,99)
     Exiv2::BasicIo::UniquePtr temporary() const {
         return Exiv2::BasicIo::UniquePtr(nullptr);
-- 
2.21.0

